streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -851,11 +853,19 @@ void runOnce() {
 
             do {
                 for (int i = 0; i < numIterations; i++) {
+                    advanceNowAndComputeLatency();
                     processed = taskManager.process(now);
 
                     if (processed > 0) {
+                        // It makes no difference to the outcome of these metrics when we record "0",
+                        // so we can just avoid the method call when we didn't process anything.
+                        processRateSensor.record(processed, now);
+
+                        // This metric is scaled to represent the _average_ processing time of _each_
+                        // task. Note, it's hard to interpret this as defined, but we would need a KIP
+                        // to change it to simply report the overall time spent processing all tasks.
```

Comment:
not a suggestion: We have a new metrics proposed in KIP-444 to report the ratio of processing each task within a thread (CPU wise), hopefully that helps with the debugging purposes.

Reply:
Yeah, I think the task-level latency metric would be the most useful. At the thread-level, it seems better to just report the total latency of all processing, so you could help debug the sources of poll latency or whatever, but it didn't seem like a good idea to just change it with no KIP.